### PR TITLE
(v7) Add MongoDB Atlas guide

### PR DESCRIPTION
**Pull request by r0mant**

### CODE REVIEW

**Comment by benarent** [comment]
This is a really cool feature. It's great to see Database Access work with DBaaS.

**Review by klizhentas — APPROVED** [review]

File: docs/pages/database-access/guides/mongodb-atlas.mdx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+Start Teleport Database service:
+
+```bash
+teleport db start \
+  --token=/tmp/token \
+  --auth-server=teleport.example.com:3080 \
+  --name=mongodb-atlas \
+  --protocol=mongodb \
+  --uri=mongodb+srv://cluster0.abcde.mongodb.net \
+  --ca-cert=/path/to/letsencrypt/root-ca
```
Comment (benarent) [inline docs/pages/database-access/guides/mongodb-atlas.mdx:38]: Suggesting we the filename the same as LetsEncrypt naming. 

```suggestion
  --ca-cert=/path/to/letsencrypt/isrgrootx1.pem
```

File: docs/pages/database-access/guides/mongodb-atlas.mdx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+### Enable self-managed X.509 authentication
+
+Teleport will authenticate with MongoDB Atlas using
+[self-managed X.509 authentication](https://docs.atlas.mongodb.com/security-self-managed-x509/).
+
+First, obtain Teleport CA certificate by running the following `tctl auth sign`
+command against your Teleport cluster:
+
+```bash
+tctl auth sign --format=mongodb --host=mongo --out=mongo
```
Comment (benarent) [inline docs/pages/database-access/guides/mongodb-atlas.mdx:100]: This works, but it's unclear with the `-ttl` is ?

Comment (r0mant) [reply docs/pages/database-access/guides/mongodb-atlas.mdx:100]: @benarent TTL flag isn't needed for this command. For Atlas we're only using CA certificate, database cert/key which you configure on-prem db with aren't used with Atlas. I can mention here that `--ttl` flag also doesn't matter.

Comment (benarent) [reply docs/pages/database-access/guides/mongodb-atlas.mdx:100]: That's a good point, and your note below makes --host info clear.

Comment (benarent) [reply docs/pages/database-access/guides/mongodb-atlas.mdx:100]: We do output this, when we run this command. Can we add a note that if using Atlas, what steps are needed vs providing the steps for self hosted mongo

```
➜  ~ tctl auth sign --format=mongodb --host=mongo --out=mongo
Database credentials have been written to mongo.crt, mongo.cas.

To enable mutual TLS on your MongoDB server, add the following to its
mongod.yaml configuration file:

net:
  tls:
    mode: requireTLS
    certificateKeyFile: /path/to/mongo.crt
    CAFile: /path/to/mongo.cas
```

File: docs/pages/database-access/guides/mongodb-atlas.mdx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+MongoDB Atlas uses certificates signed by Let's Encrypt as described in
+[Which certificate authority signs MongoDB Atlas cluster TLS certificates?](https://docs.atlas.mongodb.com/reference/faq/security/#which-certificate-authority-signs-mongodb-atlas-cluster-tls-certificates-)
+
+Download Let's Encrypt [root certificate](https://letsencrypt.org/certs/isrgrootx1.pem.txt)
+and use it as a CA in the database service configuration:
+
+```txt
+--ca-cert=/tmp/isrgrootx1.pem
+```
```
Comment (benarent) [inline docs/pages/database-access/guides/mongodb-atlas.mdx:83]: I've tried to set this with teleport.yaml and I get this error ` field ca-cert not found in type config.Database` 

My setup looks like this

```
...
            - name: "mongodb"
              protocol: "mongodb"
              uri: "mongodb+srv://atlas-mongo-teleport.f8rg9.mongodb.net"
              ca-cert: "/etc/letsencrypt/isrgrootx1.pem"
              static_labels:
                env: dev-1
                engine: 🍃🌍
```

File: docs/pages/database-access/guides/mongodb-atlas.mdx (hunk 2 above)
Comment (r0mant) [reply docs/pages/database-access/guides/mongodb-atlas.mdx:100]: Good call, I'll make a ticket and fix.

File: docs/pages/database-access/guides/mongodb-atlas.mdx (hunk 3 above)
Comment (r0mant) [reply docs/pages/database-access/guides/mongodb-atlas.mdx:83]: @benarent The yaml configuration field is called `ca_cert_file` (which is consistent naming with other services in yaml config). See the yaml reference here: https://goteleport.com/docs/database-access/reference/configuration/.

Comment (benarent) [reply docs/pages/database-access/guides/mongodb-atlas.mdx:83]: Thanks Roman, could we add an inline reference on this page for people following this guide.

Comment (r0mant) [reply docs/pages/database-access/guides/mongodb-atlas.mdx]: Good idea, added.

**Review by benarent — APPROVED** [review]
Looks good. I was able to set this up; with a few UX tweaks that I've sent over to Roman.

**Pull request merged**